sensors/imu/lsm9dsxx.c

```diff
@@ -92,9 +196,47 @@ static int lsm9dsxx_start(sensor_info_t *info)
 
 static int lsm9dsxx_alloc(sensor_info_t *info, const char *args)
 {
+	int err = 0, ret = 0;
+
 	info->types = SENSOR_TYPE_ACCEL | SENSOR_TYPE_GYRO;
 
-	/* TODO: parse arguments and initialize device */
+	/* dummy search for spi parameters in path */
+	args = strstr(args, "spi");
+	if (args != NULL && strlen(args) == strlen("spiX.X") && 0) {
+		common.spiBus = atoi(&args[3]);
+		common.spiSS = atoi(&args[5]);
+	}
+	else {
+		common.spiBus = 1;
+		common.spiSS = 0;
+	}
+	common.ctx.mode = SPI_MODE3;
+	common.ctx.speed = 6250000;
+
+	/* open SPI context */
+	err = spimsg_open(common.spiBus, common.spiSS, &common.ctx);
+	if (err < 0) {
+		printf("lsm9dsxx: failed to open SPI message context, err: %s\n", strerror(err));
+		return -1;
+	}
+
+	/* auto increment + SW reset */
+	err = (spiWriteReg(REG_CTRL_REG8, 0x05) < 0);
```

Comment:
It would be more readable to do it this way:
```suggestion
	if (spiWriteReg(REG_CTRL_REG8, 0x05) < 0)
		err = 1;
```